analysis_driver/pipelines/variant_calling.py

```diff
@@ -84,15 +84,15 @@ def filter_snp_vcf(self):
 
     @property
     def dbsnp(self):
-        dbsnp = cfg.query('genomes', self.dataset.genome_version, 'dbsnp')
-        if not dbsnp and self.dataset.pipeline.name == 'variant_calling':
-            raise AnalysisDriverError('Could not find dbsnp file for %s' % self.dataset.name)
+        dbsnp = None
+        try:
+            dbsnp = cfg.get('genomes_dir', '') + \
+                    rest_communication.get_document('genomes', where={'assembly_name': self.dataset.genome_version})['data_files']['variation']
```

Comment:
I think I would prefer if all these `rest_communication` calls where all in `dataset.py`. You can cache the result of the query and even reconstruct the full path there. This will avoid further code duplication.

Reply:
Good idea @tcezard. I'm going to create a function which does all this.